File: test/corpus/component/08-environment-force-color/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	fmt.Println("--- Environment Detection: FORCE_COLOR Test ---")
	
	// Create output with default settings
	out := termenv.NewOutput(nil)
	
	// Test color profile detection with FORCE_COLOR
	colorProfile := termenv.ColorProfile()
	fmt.Printf("ColorProfile(): %d\n", int(colorProfile))
	
	// Test environment-based color profile
	envProfile := termenv.EnvColorProfile()
	fmt.Printf("EnvColorProfile(): %d\n", int(envProfile))
	
	// Test EnvNoColor (should be false with FORCE_COLOR)
	noColor := termenv.EnvNoColor()
	fmt.Printf("EnvNoColor(): %t\n", noColor)
	
	// Test color rendering with FORCE_COLOR set
	red := out.String("Red Text").Foreground(termenv.RGBColor("#FF0000"))
	blue := out.String("Blue Text").Foreground(termenv.RGBColor("#0000FF"))
	green := out.String("Green Text").Foreground(termenv.RGBColor("#00FF00"))
	
	fmt.Printf("Red Text: %s\n", red)
	fmt.Printf("Blue Text: %s\n", blue)
	fmt.Printf("Green Text: %s\n", green)
	
	// Test with ANSI colors
	ansiRed := out.String("ANSI Red").Foreground(termenv.ANSIColor(1))
	ansiBlue := out.String("ANSI Blue").Foreground(termenv.ANSIColor(4))
	
	fmt.Printf("ANSI Red: %s\n", ansiRed)
	fmt.Printf("ANSI Blue: %s\n", ansiBlue)
	
	// Test with ANSI256 colors
	ansi256Red := out.String("ANSI256 Red").Foreground(termenv.ANSI256Color(196))
	ansi256Blue := out.String("ANSI256 Blue").Foreground(termenv.ANSI256Color(21))
	
	fmt.Printf("ANSI256 Red: %s\n", ansi256Red)
	fmt.Printf("ANSI256 Blue: %s\n", ansi256Blue)
	
	// Test styling
	bold := out.String("Bold Text").Bold()
	italic := out.String("Italic Text").Italic()
	
	fmt.Printf("Bold Text: %s\n", bold)
	fmt.Printf("Italic Text: %s\n", italic)
}

File: test/corpus/component/08-environment-force-color/case.ts
import {
  ANSI256Color,
  ANSIColor,
  ColorProfile,
  EnvColorProfile,
  EnvNoColor,
  NewOutput,
  RGBColor,
} from '../../../../src/go-style.js';

console.log('--- Environment Detection: FORCE_COLOR Test ---');

// Create output with default settings
const out = NewOutput(process.stdout);

// Test color profile detection with FORCE_COLOR
const profile = ColorProfile();
console.log(`ColorProfile(): ${profile}`);

// Test environment-based color profile
const envProfile = EnvColorProfile();
console.log(`EnvColorProfile(): ${envProfile}`);

// Test EnvNoColor (should be false with FORCE_COLOR)
const noColor = EnvNoColor();
console.log(`EnvNoColor(): ${noColor}`);

// Test color rendering with FORCE_COLOR set
const red = out.String('Red Text').Foreground(RGBColor('#FF0000'));
const blue = out.String('Blue Text').Foreground(RGBColor('#0000FF'));
const green = out.String('Green Text').Foreground(RGBColor('#00FF00'));

console.log(`Red Text: ${red.String()}`);
console.log(`Blue Text: ${blue.String()}`);
console.log(`Green Text: ${green.String()}`);

// Test with ANSI colors
const ansiRed = out.String('ANSI Red').Foreground(ANSIColor(1));
const ansiBlue = out.String('ANSI Blue').Foreground(ANSIColor(4));

console.log(`ANSI Red: ${ansiRed.String()}`);
console.log(`ANSI Blue: ${ansiBlue.String()}`);

// Test with ANSI256 colors
const ansi256Red = out.String('ANSI256 Red').Foreground(ANSI256Color(196));
const ansi256Blue = out.String('ANSI256 Blue').Foreground(ANSI256Color(21));

console.log(`ANSI256 Red: ${ansi256Red.String()}`);
console.log(`ANSI256 Blue: ${ansi256Blue.String()}`);

// Test styling
const bold = out.String('Bold Text').Bold();
const italic = out.String('Italic Text').Italic();

console.log(`Bold Text: ${bold.String()}`);
console.log(`Italic Text: ${italic.String()}`);
